apps/server/src/modules/user-import/uc/user-import.uc.ts

```diff
@@ -239,16 +286,41 @@ export class UserImportUc {
 		await this.schoolService.save(school);
 	}
 
+	private async checkSchoolMigrated(schoolId: EntityId, school: LegacySchoolDo): Promise<void> {
+		const userLoginMigration: UserLoginMigrationDO | null = await this.userLoginMigrationService.findMigrationBySchool(
+			schoolId
+		);
+
+		if (!userLoginMigration) {
+			throw new NotFoundLoggableException('UserLoginMigration', { schoolId });
+		}
+
+		if (!school.systems?.includes(userLoginMigration.targetSystemId)) {
+			throw new SchoolNotMigratedLoggableException(schoolId);
+		}
+	}
+
 	async endSchoolInMaintenance(currentUserId: EntityId): Promise<void> {
 		const currentUser = await this.getCurrentUser(currentUserId, Permission.SCHOOL_IMPORT_USERS_MIGRATE);
 		const school: LegacySchoolDo = await this.schoolService.getSchoolById(currentUser.school.id);
+
 		this.checkFeatureEnabled(school);
+
 		if (school.inUserMigration !== false || !school.inMaintenanceSince || !school.externalId) {
 			this.logger.warning(new MigrationMayNotBeCompleted(school.inUserMigration));
 			throw new BadRequestException('Sync cannot be activated for school');
 		}
+
 		school.inMaintenanceSince = undefined;
+
+		// Make the migration wizard restartable, when using the user login migration
+		const useWithUserLoginMigration: boolean = this.userImportFeatures.instance === 'n21';
```

Comment:
CPoV: please move this functionality in a private function called isMigrationRestartable and delete the comment